lnwallet/btcwallet/btcwallet.go

```diff
@@ -431,42 +431,61 @@ func (b *BtcWallet) PublishTransaction(tx *wire.MsgTx) error {
 	if err := b.wallet.PublishTransaction(tx); err != nil {
 		switch b.chain.(type) {
 		case *chain.RPCClient:
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already been spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Transaction is spending either output that is
+			// missing or already spent.
 			if strings.Contains(err.Error(), "orphan transaction") {
-				// Transaction is spending either output that
-				// is missing or already spent.
+				return lnwallet.ErrDoubleSpend
+			}
+
+			// RBF enabled transaction was rejected for whatever
+			// reason.
+			if strings.Contains(err.Error(), "replacement transaction") {
 				return lnwallet.ErrDoubleSpend
 			}
 
 		case *chain.BitcoindClient:
+			// Output was spent by other transaction already in the
+			// mempool.
 			if strings.Contains(err.Error(), "txn-mempool-conflict") {
-				// Output was spent by other transaction
-				// already in the mempool.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// RBF enabled transaction did not have enough fee.
 			if strings.Contains(err.Error(), "insufficient fee") {
-				// RBF enabled transaction did not have enough fee.
 				return lnwallet.ErrDoubleSpend
 			}
+
+			// Transaction is spending either output that is
+			// missing or already spent.
 			if strings.Contains(err.Error(), "Missing inputs") {
-				// Transaction is spending either output that
-				// is missing or already spent.
 				return lnwallet.ErrDoubleSpend
 			}
 
 		case *chain.NeutrinoClient:
+			// Output was already spent.
 			if strings.Contains(err.Error(), "already spent") {
-				// Output was already spent.
 				return lnwallet.ErrDoubleSpend
 			}
 
+			// RBF enabled transaction was rejected for whatever
+			// reason.
+			if strings.Contains(err.Error(), "replacement transaction") {
+				return lnwallet.ErrDoubleSpend
+			}
+
+			// TODO(halseth): neutrino errors depend on connected
+			// peers. Makes more sense to make it match on a
+			// combination of the btcd and bitcoind errors?
```

Comment:
Yep, this is what's done in the btcwallet and neutrino repos. I would add them as part of this PR.

Reply:
Yeah, it isn't very scalable to keep maintaining this error handling both places. Should we instead define a set of errors in btcwallet (like ErrDoubleSpend), then move all error matching there?

Reply:
Yeah for a while we've been wanting to shove all the error string matching into `btcwallet` itself, then define a set of errors on the `WalletController` interface level. This way, in `lnd` we just use `Is/As` to check for errors and we can maintain all our error matching in a single place (as there're a few areas that we duplicate this type of code between `lnd` and `btcwallet`).

Reply:
Made btcwallet PR: https://github.com/btcsuite/btcwallet/pull/647

And first use of `errors.As` 😬